app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/navigation/EditCoInsuredDestinations.kt

```diff
@@ -4,4 +4,7 @@ import com.kiwi.navigationcompose.typed.Destination
 import kotlinx.serialization.Serializable
 
 @Serializable
-data class EditCoInsuredDestination(val contractId: String) : Destination
+data class EditCoInsuredDestination(
+  val contractId: String,
+  val allowEdit: Boolean,
```

Comment:
Hmm I'd like to challenge whether we want to have a `allowEdit` on the same destination, or if we can instead have two distinct destinations, where both of them can potentially share some of the composables that they use.

My concern here is that the non editable and the editable versions of this screen may steer away from each other enough that having both in the same place might become quite hard to reason about? Reminds me of some older screens we've had which were clearly something like "either X or Y state" but those states ended up being different enough that completely separate screens would make more sense anyway.

Reply:
I agree. Should design this as two different destinations. Ill fix